test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
 	}
 }
 
+// TestMTLSWithCRLsCerts includes all the certificates needed for a particular test case of TestMTLSWithCRLs.
+type TestMTLSWithCRLsCerts struct {
+	// CABundle is the complete PEM-encoded list of certificates that will be used both by HAProxy for client
+	// validation, and by openshift-router for CRL distribution points.
+	CABundle []string
+	// CRLs is a map of the PEM-encoded CRLs, indexed by the filename that will be used in the CRL host pod's configmap.
+	CRLs map[string]string
+	// ClientCerts contains maps of the client certificates used to verify that mTLS is working as intended.
+	ClientCerts struct {
+		// Accepted is a map containing the client keys and certificates that should be able to connect to backends
+		// successfully, indexed by a unique name.
+		Accepted map[string]KeyCert
+		// Rejected is a map containing the client keys and certificates that should NOT be able to connect to backends,
+		// indexed by a unique name.
+		Rejected map[string]KeyCert
+	}
+}
+
+// TestMTLSWithCRLs verifies that mTLS works when the client auth chain includes certificate revocation lists (CRLs).
+func TestMTLSWithCRLs(t *testing.T) {
+	t.Parallel()
+	namespaceName := "mtls-with-crls"
```

Comment:
We've started to use unique namespace names. Here's an example: https://github.com/openshift/cluster-ingress-operator/blob/992b43b3cf3e1784bfe8d3083229c7ecb410e7e3/test/e2e/route_metrics_test.go#L165